.github/workflows/actions.yml

```diff
@@ -6,6 +6,20 @@ on:
     - master
 
 jobs:
+  dependency-guard:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Block dependencies from being added to package.json
+        run: |
+          IFS=', ' read -r allowed <<< "native-promise-only, zendesk_app_framework_sdk"
+          dependencies=$(npm ls --production --depth=0 --parseable | awk '{gsub(/\/.*\//,"",$1); print}')
+          
+          for dep in $dependencies; do
+            if [[ ! " ${allowed[*]} " =~ " ${dep} " ]]; then
```

Comment:
Do we need to do regexp here?

Reply:
I updated the code to do an exact match, otherwise I can replace above with below, but I found above more clear and compact

```
for dep in $dependencies; do
  found="" 
  for allow in $allowed; do if [[ ${allow} == ${dep} ]]; then found=$allow continue fi done
  if [[ -z "$found" ]]; then
    echo "::error ZAP SDK does not allow external dependencies, please remove \"${dep}\" dependency from package.json"
    exit 1
  fi
done
```

Reply:
How about

```sh
declare -A allowed
allowed=(
  ["native-promise-only"]="1"
  ["zendesk_app_framework_sdk"]="1"
)
declare -a dependencies
IFS=$'\n' read -r -d '' -a dependencies < <(npm ls --production --depth=0 --parseable --silent | awk '{gsub(/\/.*\//,"",$1); print}')

for dep in "${dependencies[@]}"; do
  if [[ -z "${allowed[$dep]}" ]]; then
    echo "::error ZAP SDK does not allow external dependencies, please remove \"${dep}\" dependency from package.json"
    exit 1
  fi
done
```

Reply:
To me they are the same, Look I am happy to copy your code in if you insist.

Reply:
I think they function the same in this particular case however I can spot a few issues in your script that go against general Bash scripting rules. For example `for dep in $dependencies`:

```sh
bash-5.1$ ls
hello.txt       world.txt
bash-5.1$ declare -a foo=('*')
bash-5.1$ for f in $foo; do printf '%s\n' "$f"; done
hello.txt
world.txt
bash-5.1$ for f in "${foo[@]}"; do printf '%s\n' "$f"; done
*
```

I understand this is irrelevant to this particular use case, but I don't see why we shouldn't hold ourselves to the good practice either. But anyway, bottom line is it works, so I'm happy with the script as it currently is.